src/shared-components/button/components/linkButton/index.tsx

```diff
@@ -22,27 +22,40 @@ const propTypes = {
   textColor: PropTypes.string,
 };
 
-const defaultProps = {
-  as: 'a',
-  buttonColor: COLORS.primary,
-  buttonType: 'primary',
-  disabled: false,
-  onClick: () => undefined,
-  textColor: '',
-};
-
 type LinkProps = {
+  /**
+   * Specifies the tag or element to be rendered
+   */
   as?: 'a' | React.ElementType;
+  /**
+   * TODO-TS: Limit type from string to COLORS constants options
+   */
   buttonColor?: string;
```

Comment:
The diff is already very large, so I largely kept the API of the `src` files the same.